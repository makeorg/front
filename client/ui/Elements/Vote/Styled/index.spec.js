import React from 'react';
import renderer from 'react-test-renderer';
import { ButtonStyle, UnvoteButtonStyle, VoteButtonStyle } from './index';

describe('ButtonStyle', () => {
  it('snapshot by default', () => {
    const component = renderer.create(<ButtonStyle />).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('must set rotate from props', () => {
    const component = renderer.create(<ButtonStyle rotate={90} />).toJSON();
    expect(component).toHaveStyleRule('transform', 'rotate(90deg)');
  });

  it('must not set rotate from props', () => {
    const component = renderer.create(<ButtonStyle rotate={0} />).toJSON();
    expect(component).not.toHaveStyleRule('transform', 'rotate(0deg)');
  });

  it('must set border-color from props', () => {
    const component = renderer.create(<ButtonStyle color="red" />).toJSON();
    expect(component).toHaveStyleRule('border-color', 'red');
  });
});

describe('UnvoteButtonStyle', () => {
  it('snapshot by default', () => {
    const component = renderer.create(<UnvoteButtonStyle />).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('must set background-color from props', () => {
    const component = renderer
      .create(<UnvoteButtonStyle color="red" />)
      .toJSON();
    expect(component).toHaveStyleRule('background-color', 'red');
  });
});

describe('VoteButtonStyle', () => {
  it('snapshot by default', () => {
    const component = renderer.create(<VoteButtonStyle />).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('must set color from props', () => {
    const component = renderer.create(<VoteButtonStyle color="red" />).toJSON();
    expect(component).toHaveStyleRule('color', 'red');
    expect(component).toHaveStyleRule('background-color', 'red', {
      modifier: ':focus',
    });
    expect(component).toHaveStyleRule('background-color', 'red', {
      modifier: ':active',
    });
  });
});
