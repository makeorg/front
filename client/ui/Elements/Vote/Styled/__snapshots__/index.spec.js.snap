// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonStyle snapshot by default 1`] = `
.c0 {
  position: relative;
  z-index: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 48px;
  height: 48px;
  border-width: 2px;
  font-size: 22px;
  border-style: solid;
  border-radius: 50%;
  overflow: hidden;
}

.c0 svg {
  position: absolute;
  top: 0;
  left: 0;
  z-index: 0;
  -webkit-transform: translate(50%,50%);
  -ms-transform: translate(50%,50%);
  transform: translate(50%,50%);
}

@media (min-width:767px) {
  .c0 {
    width: 58px;
    height: 58px;
    font-size: 26px;
  }
}

<button
  className="c0"
/>
`;

exports[`IsVotedButtonStyle snapshot by default 1`] = `
.c0 {
  position: relative;
  z-index: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 48px;
  height: 48px;
  border-width: 2px;
  font-size: 22px;
  border-style: solid;
  border-radius: 50%;
  overflow: hidden;
  color: rgb(255,255,255);
}

.c0 svg {
  position: absolute;
  top: 0;
  left: 0;
  z-index: 0;
  -webkit-transform: translate(50%,50%);
  -ms-transform: translate(50%,50%);
  transform: translate(50%,50%);
}

.c0 svg {
  fill: rgb(255,255,255);
}

@media (min-width:767px) {
  .c0 {
    width: 58px;
    height: 58px;
    font-size: 26px;
  }
}

<button
  className="sc-1gs43wt-0 c0"
/>
`;

exports[`VoteButtonStyle snapshot by default 1`] = `
.c0 {
  position: relative;
  z-index: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 48px;
  height: 48px;
  border-width: 2px;
  font-size: 22px;
  border-style: solid;
  border-radius: 50%;
  overflow: hidden;
  background-color: rgb(255,255,255);
}

.c0 svg {
  position: absolute;
  top: 0;
  left: 0;
  z-index: 0;
  -webkit-transform: translate(50%,50%);
  -ms-transform: translate(50%,50%);
  transform: translate(50%,50%);
}

.c0:focus,
.c0:active {
  color: rgb(255,255,255);
}

.c0:focus svg,
.c0:active svg {
  fill: rgb(255,255,255);
}

@media (min-width:767px) {
  .c0 {
    width: 58px;
    height: 58px;
    font-size: 26px;
  }
}

<button
  className="sc-1gs43wt-0 c0"
/>
`;
